tests/core/pyspec/eth2spec/test/phase0/unittests/fork_choice/test_on_block.py

```diff
@@ -317,7 +241,151 @@ def test_on_block_outside_safe_slots_but_finality(spec, state):
+    # Initialization
+    store = get_genesis_forkchoice_store(spec, state)
+    time = 0
+    spec.on_tick(store, time)
+
+    # Process state
+    next_epoch(spec, state)
+    _, signed_blocks, state = next_epoch_with_attestations(spec, state, False, True)
+    for block in signed_blocks:
+        spec.on_tick(store, store.time + state.slot * spec.config.SECONDS_PER_SLOT)
+        run_on_block(spec, store, block)
+    next_epoch(spec, state)
+    for _ in range(2):
+        _, signed_blocks, state = next_epoch_with_attestations(spec, state, False, True)
+        for block in signed_blocks:
+            spec.on_tick(store, store.time + state.slot * spec.config.SECONDS_PER_SLOT)
+            run_on_block(spec, store, block)
+
+    assert state.finalized_checkpoint.epoch == store.finalized_checkpoint.epoch == 0
+    assert state.current_justified_checkpoint.epoch == store.justified_checkpoint.epoch == 3
+    assert store.justified_checkpoint.hash_tree_root() == state.current_justified_checkpoint.hash_tree_root()
+
+    # Create another chain
+    # another_state = store.block_states[store.justified_checkpoint.root].copy()
+    next_epoch(spec, another_state)
+    spec.on_tick(store, store.time + another_state.slot * spec.config.SECONDS_PER_SLOT)
+
+    all_blocks = []
+    for _ in range(3):
+        _, signed_blocks, another_state = next_epoch_with_attestations(spec, another_state, True, True)
+        all_blocks += signed_blocks
+
+    assert another_state.finalized_checkpoint.epoch == 2
+    assert another_state.current_justified_checkpoint.epoch == 3
+    assert state.finalized_checkpoint.hash_tree_root() != another_state.finalized_checkpoint.hash_tree_root()
+    assert (
+        state.current_justified_checkpoint.hash_tree_root()
+        != another_state.current_justified_checkpoint.hash_tree_root()
+    )
+
+    pre_store_justified_checkpoint_root = store.justified_checkpoint.root
+    for block in all_blocks:
+        run_on_block(spec, store, block)
+
+    finalized_slot = spec.compute_start_slot_at_epoch(store.finalized_checkpoint.epoch)
+    ancestor_at_finalized_slot = spec.get_ancestor(store, pre_store_justified_checkpoint_root, finalized_slot)
+    assert ancestor_at_finalized_slot != store.finalized_checkpoint.root
+
+    assert store.finalized_checkpoint.hash_tree_root() == another_state.finalized_checkpoint.hash_tree_root()
+    assert store.justified_checkpoint.hash_tree_root() == another_state.current_justified_checkpoint.hash_tree_root()
+
+
+@with_all_phases
+@spec_state_test
+def test_new_finalized_slot_is_justified_checkpoint_ancestor(spec, state):
+    """
+    J: Justified
+    F: Finalized
+    pre-store:
+        epoch
+        [0] <- [1] <- [2] <- [3] <- [4] <- [5]
+                       F             J
+
+    another_state:
+                                 <- [4] <- [5]
+                                    F+J
+    """
+    # Initialization
+    store = get_genesis_forkchoice_store(spec, state)
+    time = 0
```

Comment:
us setting this to 0 is pretty weird because `genesis_time` is set to whatever is in `state`
which I just looked up and is default to 0... ha

still I don't think doing an `on_tick` to a time that is pulled out of thin air is not a good practice for our tests. time and ticks should be in relation to `genesis_time` and `slot` imo